File: leetcode-new/121.cpp
class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        int res = 0;
        int maxx = 0;
        for (int i = prices.size() - 1; i >= 0; i--)
        {
            res = max(res, maxx - prices[i]);
            maxx = max(maxx, prices[i]);
        }
        return res;
    }
};

File: leetcode-venture/121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        right = 1
        maxprofit = 0

        while left < right < len(prices):
            if prices[left] >= prices[right]:
                left = right
                right += 1
            else:
                maxprofit = max(maxprofit, prices[right] - prices[left])
                right += 1

        return maxprofit
